pkg/audit/stats_reporter.go

```
@@ -0,0 +1,95 @@
+package audit
+
+import (
+	"context"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/metrics"
+	"github.com/open-policy-agent/gatekeeper/pkg/util"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	totalViolationsName = "total_violations"
+	auditDurationName   = "audit_duration_seconds"
+)
+
+var (
+	violationsTotalM = stats.Int64(totalViolationsName, "Total number of violations per constraint", stats.UnitDimensionless)
+	auditDurationM   = stats.Float64(auditDurationName, "Latency of audit operation in seconds", stats.UnitSeconds)
+
+	enforcementActionKey = tag.MustNewKey("enforcement_action")
+)
+
+func init() {
+	register()
+}
+
+func register() {
+	views := []*view.View{
+		{
+			Name:        totalViolationsName,
+			Measure:     violationsTotalM,
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{enforcementActionKey},
+		},
+		{
+			Name:        auditDurationName,
+			Measure:     auditDurationM,
+			Aggregation: view.Distribution(0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1, 2, 3, 4, 5),
+		},
+	}
+
+	if err := view.Register(views...); err != nil {
+		panic(err)
```

let's return this error to init and panic inside of init().